iam/snippets/src/test/java/ServiceAccountTests.java

```diff
@@ -78,139 +76,295 @@ public void tearDown() {
   }
 
   @Test
-  public void stage1_testServiceAccountCreate() throws IOException {
-    ServiceAccount serviceAccount = CreateServiceAccount
-            .createServiceAccount(PROJECT_ID, SERVICE_ACCOUNT);
-    String got = bout.toString();
-    assertThat(got, containsString("Created service account: " + SERVICE_ACCOUNT));
-    assertNotNull(serviceAccount);
-    assertThat(serviceAccount.getName(), containsString(SERVICE_ACCOUNT));
-
+  public void testServiceAccount_createServiceAccount() throws IOException {
+    // Prepare
+    String serviceAccountName = generateServiceAccountName();
+
+    try {
+      // Act
+      ServiceAccount serviceAccount =
+          CreateServiceAccount.createServiceAccount(PROJECT_ID, serviceAccountName);
+      String got = bout.toString();
+
+      // Assert
+      assertThat(got, containsString("Created service account: " + serviceAccountName));
+      assertNotNull(serviceAccount);
+      assertThat(serviceAccount.getName(), containsString(serviceAccountName));
+    } finally {
+      // Cleanup
+      Util.tearDownTest_deleteServiceAccount(PROJECT_ID, serviceAccountName);
+    }
   }
 
   @Test
-  public void stage1_testServiceAccountsList() throws IOException {
-    IAMClient.ListServiceAccountsPagedResponse response =
-            ListServiceAccounts.listServiceAccounts(PROJECT_ID);
-
-    assertTrue(response.iterateAll().iterator().hasNext());
+  public void testServiceAccount_listServiceAccounts() throws IOException, InterruptedException {
+    // Prepare
+    String serviceAccountName = generateServiceAccountName();
+    try {
+      Util.setUpTest_createServiceAccount(PROJECT_ID, serviceAccountName);
+
+      // Act
+      IAMClient.ListServiceAccountsPagedResponse response =
+          ListServiceAccounts.listServiceAccounts(PROJECT_ID);
+
+      // Assert
+      assertTrue(response.iterateAll().iterator().hasNext());
+    } finally {
+      // Cleanup
+      Util.tearDownTest_deleteServiceAccount(PROJECT_ID, serviceAccountName);
+    }
   }
 
   @Test
-  public void stage2_testServiceAccountRename() throws IOException {
-    String renameTo = "your-new-display-name";
-    ServiceAccount serviceAccount = RenameServiceAccount
-            .renameServiceAccount(PROJECT_ID, SERVICE_ACCOUNT, renameTo);
-    String got = bout.toString();
-    assertThat(got, containsString("Updated display name"));
-    assertThat(got, containsString(renameTo));
-    assertNotNull(serviceAccount);
-    assertThat(renameTo, containsString(serviceAccount.getDisplayName()));
+  public void testServiceAccount_getServiceAccount() throws IOException, InterruptedException {
+    // Prepare
+    String serviceAccountName = generateServiceAccountName();
+    try {
+      Util.setUpTest_createServiceAccount(PROJECT_ID, serviceAccountName);
```

Comment:
nit: consider using a known SA in order to avoid creating and deleting a new SA on each execution.